#!/bin/sh /usr/share/dpatch/dpatch-run
## 02-fix-ftbfs-with-gcc-4.3.dpatch by Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
##
## DP: Fix missing includes, needed for gcc-4.3 transition.

@DPATCH@

--- a/src/a2m.cpp
+++ b/src/a2m.cpp
@@ -33,6 +33,8 @@
 
 #include "a2m.h"
 
+#include <cstring>
+
 const unsigned int Ca2mLoader::MAXFREQ = 2000,
 Ca2mLoader::MINCOPY = ADPLUG_A2M_MINCOPY,
 Ca2mLoader::MAXCOPY = ADPLUG_A2M_MAXCOPY,
--- a/src/bmf.cpp
+++ b/src/bmf.cpp
@@ -43,6 +43,8 @@
 #include "bmf.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CxadbmfPlayer::bmf_adlib_registers[117] =
 {
   0x20, 0x23, 0x40, 0x43, 0x60, 0x63, 0x80, 0x83, 0xA0, 0xB0, 0xC0, 0xE0, 0xE3,
--- a/src/cff.cpp
+++ b/src/cff.cpp
@@ -28,6 +28,8 @@
 
 #include "cff.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CcffLoader::factory(Copl *newopl)
--- a/src/dro.cpp
+++ b/src/dro.cpp
@@ -27,6 +27,8 @@
 
 #include "dro.h"
 
+#include <cstring>
+
 /*** public methods *************************************/
 
 CPlayer *CdroPlayer::factory(Copl *newopl)
--- a/src/dtm.cpp
+++ b/src/dtm.cpp
@@ -24,6 +24,8 @@
 
 #include "dtm.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CdtmLoader::factory(Copl *newopl)
--- a/src/fmc.cpp
+++ b/src/fmc.cpp
@@ -21,6 +21,8 @@
 
 #include "fmc.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CfmcLoader::factory(Copl *newopl)
--- a/src/mad.cpp
+++ b/src/mad.cpp
@@ -21,6 +21,8 @@
 
 #include "mad.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CmadLoader::factory(Copl *newopl)
--- a/src/mkj.cpp
+++ b/src/mkj.cpp
@@ -24,6 +24,8 @@
 #include "mkj.h"
 #include "debug.h"
 
+#include <cstring>
+
 CPlayer *CmkjPlayer::factory(Copl *newopl)
 {
   return new CmkjPlayer(newopl);
--- a/src/msc.cpp
+++ b/src/msc.cpp
@@ -24,6 +24,8 @@
 #include "msc.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CmscPlayer::msc_signature [MSC_SIGN_LEN] = {
   'C', 'e', 'r', 'e', 's', ' ', '\x13', ' ',
   'M', 'S', 'C', 'p', 'l', 'a', 'y', ' ' };
--- a/src/mtk.cpp
+++ b/src/mtk.cpp
@@ -21,6 +21,8 @@
 
 #include "mtk.h"
 
+#include <cstring>
+
 /*** public methods **************************************/
 
 CPlayer *CmtkLoader::factory(Copl *newopl)
--- a/src/protrack.cpp
+++ b/src/protrack.cpp
@@ -28,6 +28,8 @@
 #include "protrack.h"
 #include "debug.h"
 
+#include <cstring>
+
 #define SPECIALARPLEN	256	// Standard length of special arpeggio lists
 #define JUMPMARKER	0x80	// Orderlist jump marker
 
--- a/src/rad.cpp
+++ b/src/rad.cpp
@@ -24,6 +24,8 @@
 
 #include "rad.h"
 
+#include <cstring>
+
 CPlayer *CradLoader::factory(Copl *newopl)
 {
   return new CradLoader(newopl);
--- a/src/rat.cpp
+++ b/src/rat.cpp
@@ -32,6 +32,8 @@
 #include "rat.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CxadratPlayer::rat_adlib_bases[18] =
 {
   0x00, 0x01, 0x02, 0x08, 0x09, 0x0A, 0x10, 0x11, 0x12,
--- a/src/raw.cpp
+++ b/src/raw.cpp
@@ -21,6 +21,8 @@
 
 #include "raw.h"
 
+#include <cstring>
+
 /*** public methods *************************************/
 
 CPlayer *CrawPlayer::factory(Copl *newopl)
--- a/src/rix.cpp
+++ b/src/rix.cpp
@@ -23,6 +23,9 @@
 #include "rix.h"
 #include "debug.h"
 
+#include <cstdlib>
+#include <cstring>
+
 const unsigned char CrixPlayer::adflag[] = {0,0,0,1,1,1,0,0,0,1,1,1,0,0,0,1,1,1};
 const unsigned char CrixPlayer::reg_data[] = {0,1,2,3,4,5,8,9,10,11,12,13,16,17,18,19,20,21};
 const unsigned char CrixPlayer::ad_C0_offs[] = {0,1,2,0,1,2,3,4,5,3,4,5,6,7,8,6,7,8};
--- a/src/rol.h
+++ b/src/rol.h
@@ -25,6 +25,7 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
 
 #include "player.h"
 
--- a/src/s3m.cpp
+++ b/src/s3m.cpp
@@ -24,6 +24,8 @@
 
 #include "s3m.h"
 
+#include <cstring>
+
 const char Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
   {-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,0,1,2,3,4,5,6,7,8,-1,-1,-1,-1,-1,-1,-1};
 
--- a/src/sa2.cpp
+++ b/src/sa2.cpp
@@ -25,6 +25,8 @@
 #include "sa2.h"
 #include "debug.h"
 
+#include <cstring>
+
 CPlayer *Csa2Loader::factory(Copl *newopl)
 {
   return new Csa2Loader(newopl);
--- a/src/sng.cpp
+++ b/src/sng.cpp
@@ -21,6 +21,8 @@
 
 #include "sng.h"
 
+#include <cstring>
+
 CPlayer *CsngPlayer::factory(Copl *newopl)
 {
   return new CsngPlayer(newopl);
--- a/src/adl.cpp.old	2008-04-16 18:09:57.000000000 +0200
+++ b/src/adl.cpp	2008-04-16 18:10:21.000000000 +0200
@@ -47,10 +47,11 @@
  * $Id: adl.cpp,v 1.9 2006/08/16 00:20:45 dynamite Exp $
  *
  */
 
 #include <inttypes.h>
+#include <cstring>
 #include <stdarg.h>
 #include <assert.h>
 
 #include "adl.h"
 #include "debug.h"
